clang/test/SemaSYCL/intel-reqd-work-group-size-ast-device.cpp

```diff
@@ -77,25 +75,11 @@ class Functor16x16x16 {
   [[sycl::reqd_work_group_size(16, 16, 16)]] void operator()() const {}
 };
 
-class Functor {
-public:
-  void operator()() const {
-    f4();
-  }
-};
-
 class FunctorAttr {
 public:
   [[sycl::reqd_work_group_size(128, 128, 128)]] void operator()() const {}
 };
 
-// Test of redeclaration of [[intel::max_work_group_size()]] and [[sycl::reqd_work_group_size()]].
-[[intel::no_global_work_offset]] void func1();
-[[intel::max_work_group_size(4, 4, 4)]] void func1();
-[[sycl::reqd_work_group_size(2, 2, 2)]] void func1() {}
```

Comment:
It seems like those lines are used to check that we can apply more attributes to redeclarations and even though even if `func1` is now unused, I would still preserve those lines.

Reply:
Looking at it once more: considering that there is no propagation in SYCL 2020, applying attributes to regular device functions redeclarations doesn't make any sense. We probably could check  that attributes work properly on redeclaration of `operator()` when SYCL kernel is defined as a functor, but that seems like a separate task which is unrelated to 1.2.1 tests cleanup. I will submit a separate tracker for it and mark this thread as resolved.

Reply:
Looking at [5.8. Attributes for device code](https://registry.khronos.org/SYCL/specs/sycl-2020/html/sycl-2020.html#sec:device.attributes):

> If one of the attributes defined in this section is applied to a kernel or device function, it must be applied to the first declaration of that kernel or device function in the translation unit. Programs which fail to do this are ill formed and the compiler must issue a diagnostic. Redeclarations of the kernel or device function in the same translation unit may optionally have the same attribute applied (so long as the attribute arguments are the same between the declarations), but this is not required. The attribute remains in effect regardless of whether it appears in the redeclaration.

It seems like in SYCL 2020 there is no requirements for the attributes on redeclarations to work and we should actually emit a diagnostic if redeclaration introduces a new attribute. So, removal of this test case is a correct one for sure.

Reply:
> It seems like in SYCL 2020 there is no requirements for the attributes on redeclarations to work and we should actually emit a diagnostic if redeclaration introduces a new attribute. So, removal of this test case is a correct one for sure.

I've submitted internal tracker to review our handling of attributes on redeclarations and therefore I will mark this thread as resolved.